smyrna: fix 'path' compilation error on macOS
https://gitlab.com/graphviz/graphviz/-/issues/2315
https://gitlab.com/graphviz/graphviz/-/commit/69732da019b906551399748c2ff8a4308a29fdf4

smyrna: fix 'readlink' compilation error on macOS
https://gitlab.com/graphviz/graphviz/-/commit/ac70917bd9494a32032988e8abc7094cc9a8fa3e
--- cmd/smyrna/main.c.orig
+++ cmd/smyrna/main.c
@@ -50,7 +50,7 @@
 #include <sys/types.h>
 #endif
 
-#if !defined(__APPLE__) && !defined(_WIN32)
+#if !defined(_WIN32)
 #include <unistd.h>
 #endif
 
@@ -212,7 +212,7 @@ static char *find_me(void) {
     assert(rc != 0);
     assert(buf_size > 0);
 
-    path = gv_alloc(buf_size);
+    char *path = gv_alloc(buf_size);
 
     // retrieve the actual path
     if (_NSGetExecutablePath(path, &buf_size) < 0) {
